src/supa/connection/provider/server.py

```diff
@@ -140,24 +143,33 @@ def Reserve(self, pb_reserve_request: ReserveRequest, context: ServicerContext)
                 session.add(reservation)
                 session.flush()  # Let DB (actually SQLAlchemy) generate the connection_id for us.
                 connection_id = reservation.connection_id  # Can't reference it outside of the session, hence new var.
+                log = log.bind(connection_id=str(connection_id))
 
         # TODO modify reservation (else clause)
 
         from supa import scheduler
 
-        scheduler.add_job(ReserveJob(connection_id))
-        reserve_response = ReserveResponse(header=pb_reserve_request.header, connection_id=str(connection_id))
+        job: Job
 
-        log.info("Sending response.", response_message=reserve_response)
+        scheduler.add_job(job := ReserveJob(connection_id), trigger=job.trigger())
+        reserve_response = ReserveResponse(header=pb_reserve_request.header, connection_id=str(connection_id))
+        #
+        # FIXME: - make timeout delta configurable
+        #        - add reservation version to timeout job so we do not accidentally timeout a modify
+        #
+        log.info("Schedule reserve timeout", connection_id=str(connection_id), timeout=30)
```

Comment:
Perhaps cast the connection_id to string when you create it?

Reply:
It has its advantages to have a sticker type for connection_id and only convert it to string when we have to.